lib/cookie/cookieJar.ts

```diff
@@ -261,7 +269,9 @@ export class CookieJar {
       const parsedCookie = Cookie.parse(cookie.toString(), { loose: loose })
       if (!parsedCookie) {
         err = new Error('Cookie failed to parse')
-        return promiseCallback.reject(options?.ignoreError ? null : err)
+        return options?.ignoreError
+          ? promiseCallback.resolve(undefined)
+          : promiseCallback.reject(err)
```

Comment:
There are a lot of places where we have `ignoreError` resulting in a rejection, but that's not ignoring the error! That's just changing the root cause into a mystery.